lib/kubeclient/aws_eks_credentials.rb

```diff
@@ -38,7 +38,7 @@ def token(credentials, eks_cluster)
             'X-K8s-Aws-Id' => eks_cluster
           }
         )
-        kube_token = 'k8s-aws-v1.' + Base64.urlsafe_encode64(presigned_url_string.to_s).chomp('==')
+        kube_token = 'k8s-aws-v1.' + Base64.urlsafe_encode64(presigned_url_string.to_s).sub(/=(.*)?/, '') # rubocop:disable Metrics/LineLength
```

Comment:
Nice catch, thanks :+1:

If, for whatever reason, there is padding in the middle of the encoded data, this regexp will discard it together with everything that follows.  I don't think I ever saw that, but as [Wikipedia says](https://en.wikipedia.org/wiki/Base64#Output_padding):

> The padding character is technically not needed for decoding  ...  One exception, where padding characters are technically required, is when multiple Base64 encoded files have been concatenated. 

This really shouldn't happen here, but do you mind changing this to the obviously safe `.sub(/=*$/, '')`?

P.S. ruby 2.3+ has `Base64.urlsafe_encode64(..., padding: false)` but not 2.2 which we still support so can't use it.  That's implemented simply by `str.delete("=")`, so I guess I'm being over-pedantic.

Reply:
Done!